src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -797,8 +797,9 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return defaultProvider && this._providers.has(defaultProvider) ? defaultProvider : undefined;
 	}
 
-	public getDefaultAuthenticationTypeId(): string {
-		let defaultAuthenticationType = WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultAuthenticationType);
+	public getDefaultAuthenticationTypeId(providerName: string): string {
+		const authOption = this._capabilitiesService.getCapabilities(providerName)?.connection.connectionOptions?.find(option => option.specialValueType === ConnectionOptionSpecialType.authType);
+		const defaultAuthenticationType = authOption ? WorkbenchUtils.getSqlConfigValue<string>(this._configurationService, Constants.defaultAuthenticationType) : undefined;
```

Comment:
Shouldn't we also be checking to see if the auth type is actually supported by the provider? Right now if they only support basic as an auth type for example it'll still pull whatever value is configured there.

And if we're talking about further improvements - it also seems like this should be something set per-provider. Having a single default auth type for everyone isn't going to work long term either.

Reply:
I have made an issue for this: https://github.com/microsoft/azuredatastudio/issues/20573